elasticgraph-warehouse/lib/elastic_graph/warehouse/schema_definition/warehouse_table.rb

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/warehouse/schema_definition/field_type_converter"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      # Represents a warehouse table configuration.
+      class WarehouseTable < Struct.new(:name, :indexed_type)
```

Comment:
Can this use `::Data.define` instead?

In general, we prefer `::Data.define` over `::Struct.new` because `::Data.define` creates an immutable data class whereas `::Struct.new` makes a mutable one. Many/most classes in `elasticgraph-schema_definition` require mutability (and therefore use `::Struct`) but this one can use `::Data.define` I think--we don't need to allow `name` or `indexed_type` to be mutated.